SchemaDownlader: Update to download deprecated input fields

ndwhelan opened the pull request:
Added `(includeDeprecated: true)` as a params to `inputFields` in the `fragment` portion of the GraphQL query to download a schema.

### Current Behavior

When the `downloadApolloSchema` Gradle task is run, input fields with the `@deprecated` tag **ARE NOT** downloaded.

### Expected Change in Behavior

When the `downloadApolloSchema` Gradle task is run, input fields with the `@deprecated` tag **ARE** downloaded.

apollo-cla commented:
@ndwhelan: Thank you for submitting a pull request! Before we can merge it, you'll need to sign the Apollo Contributor License Agreement here: https://contribute.apollographql.com/

martinbonnin commented on libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/SchemaDownloader.kt:
```diff
@@ -199,7 +199,7 @@ object SchemaDownloader {
         deprecationReason
       }
-      inputFields {
+      inputFields(includeDeprecated: true {
```
Looks like you're missing a closing parenthesis here

martinbonnin commented on libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/SchemaDownloader.kt:
```diff
@@ -199,7 +199,7 @@ object SchemaDownloader {
       }
-      inputFields {
+      inputFields(includeDeprecated: true {
         ...InputValue
```
You'll also want to update the `InputValue` fragment to include `isDeprecated` and `deprecationReason`

ndwhelan replied:
Done.

ndwhelan commented:
Wow, that is embarrassing. Sorry for that. I need to actually pull the code
and do this properly instead of just using the GitHuh editor. Thank you for
the advice.

On Fri, Feb 10, 2023, 5:55 AM Martin Bonnin ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/SchemaDownloader.kt
> <https://github.com/apollographql/apollo-kotlin/pull/4678#discussion_r1102676833>
> :
>
> > @@ -199,7 +199,7 @@ object SchemaDownloader {
>          isDeprecated
>          deprecationReason
>        }
> -      inputFields {
> +      inputFields(includeDeprecated: true {
>          ...InputValue
>
> You'll also want to update the InputValue fragment to include isDeprecated
> and deprecationReason
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apollographql/apollo-kotlin/pull/4678#pullrequestreview-1293034461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHINQ6XJQTL5CKVWYVPGYLWWYUEHANCNFSM6AAAAAAUXDNG2Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

martinbonnin commented:
Hi @ndwhelan how can we help here? Want us to get this through the finish line?

ndwhelan commented:
Apologies. Numerous high priority things came up for me this week. I'd like
to try to finish it this weekend.

If you think it's preferable that you take it on at this point, though, I
certainly would not have any problem with that.

Best,

Nick

On Fri, Feb 17, 2023, 3:45 AM Martin Bonnin ***@***.***>
wrote:

> Hi @ndwhelan <https://github.com/ndwhelan> how can we help here? Want us
> to get this through the finish line?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apollographql/apollo-kotlin/pull/4678#issuecomment-1434389314>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHINQ275SJ2W4EFIGQNWN3WX5CCDANCNFSM6AAAAAAUXDNG2Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

martinbonnin commented:
No worries at all! Take your time, just wanted to make sure this doesn't fall through the cracks.

ndwhelan commented:
I've updated things from the PR, and updated a test. The tests are passing, but I wasn't really convinced it wouldn't return the whole thing regardless of my query in SchemaDownloader. I want to say it was, because it would fail if I put bad syntax in, but I'm not 100% sure. Any feedback on the test or improving that would be appreciated. I should be able to resolve it promptly if necessary.

martinbonnin reviewed (approved):
This all looks very good to me, Thanks! 🙏

BoD reviewed (approved):
👍

BoD commented:
This has been released as part of [v3.7.5](https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.5)
